### PR TITLE
Add io.github.fpiesche.f2bgl

### PR DESCRIPTION
**Pull request by fpiesche**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
  - the `device=all` permission is to allow the game's joystick/gamepad support to work out of the box.
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub.
  - The original f2bgl project is feature complete and archived; I've created my own fork to be able to add FreeDesktop metadata for a Flatpak build upstream. I have contacted the original author to clarify which license their code is under as the original repository does not specify this and will finalise the PR once I have this information.
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*
  - The only patch is to redirect the default file name for the key mapping file to allow users to remap keys (as this is normally stored with the game executable).

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

File: io.github.fpiesche.f2bgl.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            fi
+          # Launch game
+          - cd /app/bin
+          - ./f2bgl --datapath=$XDG_DATA_HOME/game --savepath=$XDG_DATA_HOME/user --subtitles $@
+    buildsystem: simple
+    subdir: src
+    builddir: true
+    build-commands:
+      - make -j$(nproc)
+      - install -Dm 744 f2bgl -t /app/bin
```
Comment (hfiguiere) [inline io.github.fpiesche.f2bgl.yml:69]:
```suggestion
      - install -Dm 755 f2bgl -t /app/bin
```
that's the proper mode

File: io.github.fpiesche.f2bgl.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+      - install -Dm 744 controls.cfg -t /app/bin
+      - install -Dm 744 ../io.github.fpiesche.f2bgl.sh -t /app/bin
```
Comment (hfiguiere) [inline io.github.fpiesche.f2bgl.yml:71]:
```suggestion
      - install -Dm 755 ../io.github.fpiesche.f2bgl.sh -t /app/bin
```

File: io.github.fpiesche.f2bgl.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+                install -Dm744 /app/bin/controls.cfg -t ${XDG_DATA_HOME}/user/;
+            fi
+          # Launch game
+          - cd /app/bin
+          - ./f2bgl --datapath=$XDG_DATA_HOME/game --savepath=$XDG_DATA_HOME/user --subtitles $@
+    buildsystem: simple
+    subdir: src
+    builddir: true
+    build-commands:
+      - make -j$(nproc)
```
Comment (hfiguiere) [inline io.github.fpiesche.f2bgl.yml:68]:
```suggestion
      - make -j$FLATPAK_BUILDER_N_JOBS
```

File: io.github.fpiesche.f2bgl.yml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+command: io.github.fpiesche.f2bgl.sh
+
+finish-args:
+  # hardware 3D
+  - --device=dri
+  # gamepad support
+  - --device=all
```
Comment (hfiguiere) [inline io.github.fpiesche.f2bgl.yml:11]:
```suggestion
  # gamepad support
  - --device=all
```

File: io.github.fpiesche.f2bgl.yml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+          # Launch game
+          - cd /app/bin
+          - ./f2bgl --datapath=$XDG_DATA_HOME/game --savepath=$XDG_DATA_HOME/user --subtitles $@
+    buildsystem: simple
+    subdir: src
+    builddir: true
+    build-commands:
+      - make -j$(nproc)
+      - install -Dm 744 f2bgl -t /app/bin
+      - install -Dm 744 controls.cfg -t /app/bin
```
Comment (hfiguiere) [inline io.github.fpiesche.f2bgl.yml:70]:
```suggestion
      - install -Dm 644 controls.cfg -t /app/bin
```

not an executable.

**Comment by hfiguiere** [comment]
bot, build io.github.fpiesche.f2bgl

**Comment by fpiesche** [comment]
bot, build io.github.fpiesche.f2bgl

**Pull request closed without merging**